kraken-java-sdk/kraken-java-sdk-sync/src/main/java/com/consoleconnect/kraken/operator/sync/service/PullDeploymentService.java

```diff
@@ -102,7 +102,12 @@ protected void ingestData(UnifiedAssetDto dto) {
 
   @Scheduled(cron = "${app.cron-job.pull-latest-release:-}")
   public void scheduledCheckLatestProductRelease() {
+    for (int i = 0; i < 5; i++) {
```

Comment:
Can I know why there is a loop here ?  and why is it  `5` ?

Reply:
The task is triggered every 30 seconds, and only one deployment can be pulled at a time, resulting in a long upgrade time.

Reply:
for the above logic,  I can't fully understand it.   I would like other team members can review it based on the feature requirement, from my perspective, there maybe a issue here, 
1.  the scheduled can be customized via app.cron-job.pull-latest-release, it maybe shorter than 30s or can be longer than 30s
2. `5` is just a idea value based on the current test, if there are some factors changed,  `5` may need to be changed as well.

Reply:
For example, a template upgrade contains 16 use cases, each of which generates a separate deployment. Therefore, the data plane actually needs 16 API calls, each with an interval of 30 seconds, which takes about 5 minutes. This time is too long, so it is allowed to loop five times in one job to reduce the upgrade time.

Reply:
mabye could consider to use an expression to caculate the times, for example: 
batchSize = (totalApiUsecases * 30 * perUserCaseCost) / expectCompleteTime

Configure the variables outside the yaml file to make this readable.